pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -163,17 +164,20 @@ func (c *Consolidation) computeConsolidation(ctx context.Context, node Candidate
 	}
 
 	return Command{
-		nodesToRemove:   []*v1.Node{node.Node},
-		action:          actionReplace,
-		replacementNode: newNodes[0],
-		created:         c.clock.Now(),
+		nodesToRemove:    []*v1.Node{node.Node},
+		action:           actionReplace,
+		replacementNodes: []*pscheduling.Node{newNodes[0]},
+		created:          c.clock.Now(),
 	}, nil
 }
 
 // validateCommand validates a command for a deprovisioner
 func (c *Consolidation) ValidateCommand(ctx context.Context, nodesToDelete []CandidateNode, cmd Command) (bool, error) {
-	if cmd.action == actionDelete && c.validateDeleteEmpty(nodesToDelete) {
-		return true, nil
+	// If we're deleting and there are empty nodes, command is valid.
+	if cmd.action == actionDelete {
```

Comment:
Why is this two separate If blocks now? Are we meant to handle the case when validating delete fails?

Reply:
Technically this should have no difference in function. What was happening here was if the action was Delete and we had any empty nodes to delete, this would return true. Deleting non-empty nodes and doing the replace action are the same except the command has a replacementNode. Since we can also delete non-empty nodes with consolidation without needing a replacement, this just delineates the logic a little more clearly in my eyes.

Reply:
Typically, you don't want to nest if statements unless there's a specific reason here. If there's no else logic here, that tells me these statements should go together. 

Won't block on it, but I disagree that this makes the code cleaner.